examples/mnist/extract_mnist.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+from __future__ import print_function
+import argparse
+import numpy
+
+import chainer
+from chainer.training import extensions
+
+import train_mnist
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Chainer example: MNIST')
+    parser.add_argument('--batchsize', '-b', type=int, default=100,
+                        help='Number of images in each mini batch')
+    parser.add_argument('--gpu', '-g', type=int, default=-1,
+                        help='GPU ID (negative value indicates CPU)')
+    parser.add_argument('--out', '-o', default='result',
+                        help='Directory to output the result')
+    parser.add_argument('--unit', '-u', type=int, default=1000,
+                        help='Number of units')
+    parser.add_argument('model', help='Load the model from snapshot')
+    args = parser.parse_args()
+
+    print('GPU: {}'.format(args.gpu))
+    print('# unit: {}'.format(args.unit))
+    print('# Minibatch-size: {}'.format(args.batchsize))
+    print('')
+
+    # Set up a neural network to load trained model into.
+    model = train_mnist.MLP(784, args.unit, 10)
```

Comment:
`args.unit` needs to be the same as was passed to `train_mnist.py`. See comment below regarding using only `train_mnist.py` for this example.